app/controllers/registrations_controller.rb

```diff
@@ -644,15 +710,18 @@ def load_payment_intent
 
   def refund_payment
     registration = Registration.find(params[:id])
-    stripe_integration = registration.competition.payment_account_for(:stripe)
+    payment = RegistrationPayment.find(params[:payment_id])
 
-    unless stripe_integration.present?
-      flash[:danger] = "You cannot emit refund for this competition anymore. Please use your Stripe dashboard to do so."
+    # FIXME: This is a bit of a hack but we're getting rid of legacy registration
+    #   refunds soon anyways, so it's fine to keep this while the old system lurks around.
+    account_type = payment.receipt.is_a?(StripeRecord) ? :stripe : :paypal
+    payment_account = registration.competition.payment_account_for(account_type)
+
+    unless payment_account.present?
+      flash[:danger] = "You cannot emit refund for this competition anymore. Please use the integration's dashboard to do so."
       return redirect_to edit_registration_path(registration)
     end
 
-    payment = RegistrationPayment.find(params[:payment_id])
-
     refund_amount_param = params.require(:payment).require(:refund_amount)
     refund_amount = refund_amount_param.to_i
```

Comment:
Why is it not just:
```ruby
refund_amount = params.require(:payment).require(:refund_amount).to_i
```

(I know it's not part of the diff, but may as well clean it up while we're here)